File: 0144_Binary Tree Preorder Traversal/BinaryTreePreorderTraversal.cs
public class Solution {
    public IList<int> PreorderTraversal (TreeNode root) {
        IList<int> ret = new List<int> ();
        if (root == null) return ret;
        Stack<TreeNode> stack = new Stack<TreeNode> ();
        stack.Push (root);
        while (stack.Count () > 0) {
            var node = stack.Pop ();
            ret.Add (node.val);
            if (node.right != null) stack.Push (node.right);
            if (node.left != null) stack.Push (node.left);
        }

        return ret;
    }
}

File: 0144_Binary Tree Preorder Traversal/BinaryTreePreorderTraversal.java
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> list = new LinkedList<Integer>();
        if(root == null) return list;
        Stack<TreeNode> stack = new Stack<TreeNode>();
        stack.push(root);
        while(!stack.isEmpty()){
            TreeNode node = stack.pop();
            if(node.right !=null) stack.push(node.right);
            if(node.left != null) stack.push(node.left);
            list.add(node.val);
        }       

        return list;
    }
}